Source/AwsCommonRuntimeKit/event-stream/EventStreamMessage.swift

```diff
@@ -47,90 +47,89 @@ public struct EventStreamMessage {
-                        rawHeaders,
-                        headerName,
-                        UInt8(header.name.count),
-                        Int8(value.uintValue))
-                case .byte(let value):
-                    return aws_event_stream_add_byte_header(
-                        rawHeaders,
-                        headerName,
-                        UInt8(header.name.count),
-                        value)
-                case .int16(let value):
-                    return aws_event_stream_add_int16_header(
-                        rawHeaders,
-                        headerName,
-                        UInt8(header.name.count),
-                        value)
-                case .int32(let value):
-                    return aws_event_stream_add_int32_header(
+            let headerNameLength = UInt8(header.name.utf8.count)
+            
+            switch header.value {
+            case .bool(let value):
+                return aws_event_stream_add_bool_header(
+                    rawHeaders,
+                    header.name,
+                    headerNameLength,
+                    Int8(value.uintValue))
+            case .byte(let value):
+                return aws_event_stream_add_byte_header(
+                    rawHeaders,
+                    header.name,
+                    headerNameLength,
+                    value)
+            case .int16(let value):
+                return aws_event_stream_add_int16_header(
+                    rawHeaders,
+                    header.name,
+                    headerNameLength,
+                    value)
+            case .int32(let value):
+                return aws_event_stream_add_int32_header(
+                    rawHeaders,
+                    header.name,
+                    headerNameLength,
+                    value)
+            case .int64(let value):
+                return aws_event_stream_add_int64_header(
+                    rawHeaders,
+                    header.name,
+                    headerNameLength,
+                    value)
+            case .byteBuf(var value):
+                if value.count > EventStreamHeader.maxValueLength {
+                    throw CommonRunTimeError.crtError(
+                        .init(
+                            code: AWS_ERROR_EVENT_STREAM_MESSAGE_INVALID_HEADERS_LEN.rawValue))
+                }
+                return value.withUnsafeMutableBytes {
+                    let bytes = $0.bindMemory(to: UInt8.self).baseAddress!
+                    return aws_event_stream_add_bytebuf_header(
                         rawHeaders,
-                        headerName,
-                        UInt8(header.name.count),
-                        value)
-                case .int64(let value):
-                    return aws_event_stream_add_int64_header(
+                        header.name,
+                        headerNameLength,
+                        bytes,
+                        UInt16($0.count),
```

Comment:
trivial: similar, we get length in 2 mildly different ways here: `value.count` up at the length check, then `$0.count` down here. Can we use a temp variable, or do `value.count` in both places, or move the length-check down so we can do`$0.count` in both places?